src/modules/olMap/ol/source/RefreshableXYZ.js

```diff
@@ -0,0 +1,26 @@
+/**
+ * @module modules/olMap/ol/source/RefreshableXYZ
+ */
+import { XYZ } from 'ol/source';
+
+/**
+ * `XYZ` source that can refresh its tiles "smoothly".
+ */
+export class RefreshableXYZ extends XYZ {
+	/**
+	 * @param {ol.Options} [options] XYZ options.
+	 */
+	constructor(options) {
+		super(options);
+	}
+	/**
+	 *
+	 * Smoothly updates the tiles by calling the protected method `setKey`.
+	 * Note: calling `changed` does nothing and calling `refresh` prunes the cache which is visually not smooth.
```

Comment:
Found this discussion (https://gis.stackexchange.com/questions/302532/how-to-update-tile-source-url-at-zoom-change), which could to be mentioned, to be helpful for future visitors of this code.